photon-api/src/main/scala/com/linkedin/photon/ml/model/RandomEffectModelInProjectedSpace.scala

```diff
@@ -37,30 +39,48 @@ protected[ml] class RandomEffectModelInProjectedSpace(
     randomEffectType,
     featureShardId) {
 
+  /**
+   *
+   * @param storageLevel The storage level
+   * @return This object with all its RDDs' storage level set
+   */
   override def persistRDD(storageLevel: StorageLevel): this.type = {
     if (!modelsInProjectedSpaceRDD.getStorageLevel.isValid) modelsInProjectedSpaceRDD.persist(storageLevel)
     this
   }
 
+  /**
+   *
+   * @return This object with all its RDDs unpersisted
+   */
   override def unpersistRDD(): this.type = {
     if (modelsInProjectedSpaceRDD.getStorageLevel.isValid) modelsInProjectedSpaceRDD.unpersist()
     this
   }
 
+  /**
+   *
+   * @param name The parent name for the model RDD in this class
+   * @return This object with all its RDDs' name assigned
+   */
   override def setName(name: String): this.type = {
     modelsInProjectedSpaceRDD.setName(name)
     this
   }
 
+  /**
+   *
+   * @return This object with all its RDDs materialized
+   */
   override def materialize(): this.type = {
     modelsInProjectedSpaceRDD.count()
     this
   }
 
   /**
-   * Build a summary string for the model
+   * Summarize this model in text format.
```

Comment:
"A human readable textual summary"? There is "text format" and "string representation": just choose one?